application/src/main/java/org/togetherjava/tjbot/moderation/ModAuditLogWriter.java

```diff
@@ -0,0 +1,152 @@
+package org.togetherjava.tjbot.moderation;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.GenericInteractionCreateEvent;
+import net.dv8tion.jda.api.requests.restaction.MessageAction;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.config.Config;
+
+import java.awt.*;
+import java.nio.charset.StandardCharsets;
+import java.time.temporal.TemporalAccessor;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+/**
+ * used to log activities in the mod audit log channel
+ */
+public enum ModAuditLogWriter {
+    ;
+    private static final Color EMBED_COLOR = Color.decode("#3788AC");
+
+    private static final Logger logger = LoggerFactory.getLogger(ModAuditLogWriter.class);
+
+    private static final Predicate<String> isAuditLogChannelName =
+            Pattern.compile(Config.getInstance().getModAuditLogChannelPattern()).asMatchPredicate();
+    private static final Predicate<TextChannel> isAuditLogChannel =
+            channel -> isAuditLogChannelName.test(channel.getName());
+
+    /**
+     * logs an entry in the mod audit log channel.
```

Comment:
* `Logs` not `logs`, correct english in javadoc please.
* The description is really sparse, please expand it a bit.